src/miner.rs

```diff
@@ -66,23 +66,24 @@ impl MinerManager {
             }
         };
 
-        for c in &self.block_channels {
-            c.send(state.clone()).await.map_err(|e| e.to_string())?;
+        if !&self.block_channel.is_closed() {
+            self.block_channel.send(state.clone()).map_err(|_e| "Failed sending block to threads")?;
         }
         Ok(())
     }
 
     fn launch_miner(
         send_channel: Sender<KaspadMessage>,
-        mut block_channel: Receiver<Option<pow::State>>,
+        block_channel: watch::Receiver<Option<pow::State>>,
         hashes_tried: Arc<AtomicU64>,
     ) -> MinerHandler {
         let mut nonce = Wrapping(thread_rng().next_u64());
         std::thread::spawn(move || {
             let mut state = None;
             loop {
-                if state.is_none() {
-                    state = block_channel.blocking_recv().ok_or(TryRecvError::Disconnected)?;
+                while state.is_none() {
+                    sleep(Duration::from_millis(500));
+                    state = block_channel.borrow().clone();
```

Comment:
You can use `block_on` from tokio, this is how the older approach worked too: https://docs.rs/tokio/latest/src/tokio/sync/mpsc/bounded.rs.html#285-287

Reply:
I can't use `tokio::future` directly because is a private module. Should I create a runtime for this? I mean:
```
# Before the loop
let mut rt = tokio::runtime::Runtime::new().unwrap();
# In the loop
rt.block_on(block_channel.changed())?;
```